halo2_proofs/tests/plonk_api.rs

```diff
@@ -383,602 +392,192 @@ fn plonk_api() {
-                let guard = f(self.msm)?;
-                let g = guard.compute_g();
-                let (msm, _) = guard.use_g(g);
-                if msm.eval() {
-                    Ok(())
-                } else {
-                    Err(Error::ConstraintSystemFailure)
-                }
-            }
-        }
+        verify_proof::<
+            _,
+            VerifierIPA<_>,
+            Blake2bRead<_, _, Challenge255<_>>,
+            BatchVerifier<_, _>,
+            _,
+            _,
+        >(rng, verifier_params, pk.get_vk(), &proof[..]);
+    }
 
-        {
-            let strategy = AccumulationVerifier::new(&params);
-            let mut transcript = Blake2bRead::<_, _, Challenge255<_>>::init(&proof[..]);
-            assert!(verify_proof(
-                &params,
-                pk.get_vk(),
-                strategy,
-                &[&[&pubinputs[..]], &[&pubinputs[..]]],
-                &mut transcript,
-            )
-            .is_ok());
-        }
+    fn test_plonk_api_gwc() {
+        use halo2_proofs::poly::kzg::commitment::{KZGCommitmentScheme, ParamsKZG};
+        use halo2_proofs::poly::kzg::multiopen::{ProverGWC, VerifierGWC};
+        use halo2_proofs::poly::kzg::strategy::BatchVerifier;
+        use halo2curves::bn256::Bn256;
 
-        //
-        // Test batch-verifier strategy.
-        //
+        let params = ParamsKZG::<Bn256>::new(K);
+        let rng = OsRng;
 
-        {
-            let strategy = BatchVerifier::new(&params, OsRng);
-
-            // First proof.
-            let mut transcript = Blake2bRead::<_, _, Challenge255<_>>::init(&proof[..]);
-            let strategy = verify_proof(
-                &params,
-                pk.get_vk(),
-                strategy,
-                &[&[&pubinputs[..]], &[&pubinputs[..]]],
-                &mut transcript,
-            )
-            .unwrap();
-
-            // "Second" proof (just the first proof again).
-            let mut transcript = Blake2bRead::<_, _, Challenge255<_>>::init(&proof[..]);
-            let strategy = verify_proof(
-                &params,
-                pk.get_vk(),
-                strategy,
-                &[&[&pubinputs[..]], &[&pubinputs[..]]],
-                &mut transcript,
-            )
-            .unwrap();
+        let pk = synthesize::<KZGCommitmentScheme<_>>(&params);
 
-            // Check the batch.
-            assert!(strategy.finalize());
-        }
+        let proof = create_proof::<_, Blake2bWrite<_, _, Challenge255<_>>, ProverGWC<_>, _, _>(
+            rng, &params, &pk,
+        );
+
+        let verifier_params = params.verifier_params();
+
+        verify_proof::<
+            _,
+            VerifierGWC<_>,
+            Blake2bRead<_, _, Challenge255<_>>,
+            BatchVerifier<_, _>,
+            _,
+            _,
+        >(rng, verifier_params, pk.get_vk(), &proof[..]);
     }
 
-    // Check that the verification key has not changed unexpectedly
```

Comment:
I think keeping this vk check would be useful, especially when we start to store vk instead of generating it on-the-fly in production, we can make sure changes to halo2 won't break the dependents.

Reply:
Restored via [e713f30](https://github.com/privacy-scaling-explorations/halo2/pull/77/commits/e713f303be9cea50bac683db20a9dce823fd72e6)